lib/api/controllers/bulkController.js

```diff
@@ -103,29 +114,46 @@ class BulkController extends BaseController {
    * @param {Request} request
    * @returns {Promise<Object>}
    */
-  mWrite(request) {
-    assertHasBody(request);
-    assertBodyHasAttribute(request, 'documents');
-    assertBodyAttributeType(request, 'documents', 'array');
-    assertHasIndexAndCollection(request);
-
-    const notify = this.tryGetBoolean(request, 'args.notify');
-
-    return this.engine.mcreateOrReplace(request, false)
-      .then(response => {
-        if (response.error.length > 0) {
+  mWrite (request) {
+    const
+      { index, collection } = this.getIndexAndCollection(request),
+      documents = this.getArrayParam(request, 'body.documents'),
+      refresh = this.getStringParam(request, 'args.refresh', 'false'),
+      notify = this.tryGetBoolean(request, 'args.notify');
+
+    return this.publicStorage.mCreateOrReplace(
+      index,
+      collection,
+      documents,
+      { refresh, injectKuzzleMeta: false }
+    )
+      .then(({ items, errors }) => {
+        if (errors.length > 0) {
           request.setError(
-            errorsManager.getError('document_creations_failed', response.error));
+            errorsManager.getError('document_creations_failed', errors));
         }
 
         if (notify) {
           this.kuzzle.notifier.notifyDocumentMChanges(
             request,
-            response.result,
+            items,
             true);
         }
 
-        return { hits: response.result, total: response.result.length };
+        const hits = [];
+
+        for (const item of items) {
+          hits.push({
+            _id: item._id,
+            _source: item._source,
+            _version: item._version,
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this trailing comma. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3ATrailingComma)